src/CommunityToolkit.Maui.Core/Services/DialogFragmentService.android.cs

```diff
@@ -68,18 +65,16 @@ static void HandleStatusBarColor(DialogFragment dialogFragment, AppCompatActivit
 
 		var statusBarColor = activity.Window.StatusBarColor;
 		var platformColor = new Android.Graphics.Color(statusBarColor);
-		var dialog = dialogFragment.Dialog;
-
-		Debug.Assert(dialog is not null);
-		Debug.Assert(dialog.Window is not null);
-
-		var window = dialog.Window;
+		if (dialogFragment.Dialog?.Window is not Window dialogWindow)
+		{
+			throw new InvalidOperationException("Dialog window cannot be null");
```

Comment:
@brminnick  we shouldn't throw here, since there's no way for devs to catch this exception. I used `Debug.Assert` because it's what the runtime/sdk uses to _safe check for null_ . But here you're checking against a type which can be false and throw this exception.

Reply:
Using [`Debug.Assert()`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert?view=net-9.0) is a bit of an anti-pattern because it only works when building in Debug configuration:

  > [Debug.Assert](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert?view=net-9.0) method works only in debug builds.

This means that when users consume our library and the value of `dialog.Window` is `null`, the following code in Release builds is guaranteed to throw an unhelpful `NullReferenceException`:

```cs
Debug.Assert(dialog is not null);
Debug.Assert(dialog.Window is not null);

var window = dialog.Window;
```

In what scenario would this code be executed called _and_ `dialog.Window` also be null? If this scenario exists, I agree that we should implement a solution.

Reply:
I think just returning is enough, the only scenario I can imagine that will happening is when the app is backgrounded or is moving into background.